api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -422,4 +423,103 @@ public void setHttpOnly(boolean isHttpOnly) {
     public boolean isHttpOnly() {
         return isHttpOnly;
     }
+
+    /**
+     * Sets the <i>SameSite</i> flag for this cookie with the provided value.
+     *
+     * <p>
+     *     When <code>sameSiteMode</code> is not <code>null</code>, this cookie will have its <code>SameSite</code>
+     *     attribute set with the provided value.
+     * </p>
+     *
+     * @param sameSiteMode The <i>SameSite</i> attribute's value. May be null if the attribute should not be set.
+     *
+     * @since Servlet 4.x
+     */
+    public void setSameSiteMode(SameSiteMode sameSiteMode) {
+        this.sameSiteMode = sameSiteMode;
+    }
+
+    /**
+     * Returns the <i>SameSite</i>> mode associated with this cookie.
+     *
+     * @return The <i>SameSite</i> mode associated with this cookie. May be null.
+     *
+     * @since Servlet 4.x
+     */
+    public SameSiteMode getSameSiteMode() {
+        return sameSiteMode;
+    }
+
+    /**
+     * The available values that can be used with the cookie's <i>SameSite</i> attribute.
+     */
+    public enum SameSiteMode {
```

Comment:
The problem with having this here is that I think we also need to be able to set the same site mode in `SessionCookieConfig`, which unfortunately is not in the `http` package.

I'm not sure how to solve this other than just letting `SessionCookieConfig` have a dependency on this enum here.   However, this PR needs a solution one way or another before it can be accepted.

Reply:
I suppose updating SessionCookieConfig would also potentially require a web.xml schema update? Is there a process of updating that?

Reply:
I can move SameSiteMode to jakarta.servlet.http.cookie package if that helps, or move it just up one level to jakarta.servlet.http package. Which would you prefer?

Reply:
Tough... I don't think there are any pretty solutions. 
However, as `SessionCookieConfig` already has javadoc references to `javax.servlet.http.Cookie` I think the best thing to do is leave it here and just use it directly from `SessionCookeConfig`

Reply:
Yes - eventually we need a schema update.  No that is not in process that I know of.   However we can change `SessionCookieConfig` without a schema change as it can be used programatically

Reply:
All of the schemas will be changed to Jakarta EE 9.
This change can be done along with the other required changes.
The schemas are all [here](https://github.com/eclipse-ee4j/jakartaee-schemas).
Submit a PR and someone in the platform project can approve it.

Reply:
I have created draft https://github.com/eclipse-ee4j/jakartaee-schemas/pull/1 and would appreciate review before I make it non draft.
I have used cased: None, Lax, Strict, again to match the RFC, so I'm wondering if we should do the same with the enum names?

Reply:
The enum names should match the Java programming language conventions.